tests/Feature/Api/Location/StoreControllerTest.php

```diff
@@ -0,0 +1,153 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\Feature\Api\Location;
+
+use App\Domain\Locations\Models\Location;
+use App\Domain\Trips\Models\Trip;
+use App\Models\User;
+use Illuminate\Foundation\Testing\WithFaker;
+use Illuminate\Testing\Fluent\AssertableJson;
+use Laravel\Sanctum\Sanctum;
+use PHPUnit\Framework\Attributes\Group;
+use PHPUnit\Framework\Attributes\Test;
+use Tests\TestCase;
+
+#[Group("Locations")]
+class StoreControllerTest extends TestCase
+{
+    use WithFaker;
+
+    #[Test]
+    public function it_wont_allow_an_unauthenticated_user_to_create_a_location(): void
+    {
+        $response = $this->postJson(route('api.locations.store'), []);
+
+        $response->assertUnauthorized();
+    }
+
+    #[Test]
+    public function it_will_fail_validation_when_trying_to_create_a_location(): void
+    {
+        $postData = [];
+
+        Sanctum::actingAs(User::factory()->create());
+
+        $response = $this->postJson(route('api.locations.store'), $postData);
+
+        $response->assertUnprocessable();
+
+        $response->assertJsonValidationErrors(
+            [
+                'latitude',
+                'longitude',
+                'name',
+                'trip_id',
+            ]
+        );
+    }
+
+    #[Test]
```

Comment:
does this work with our current coding standards? just checking if it needs updating to support this

Reply:
yeah this seems to be ok ive not had any issue coming from ECS at all here.